homeassistant/components/media_player/kodi.py

```diff
@@ -94,8 +90,8 @@ def __init__(self, hass, name, host, port, username=None, password=None,
         else:
             image_auth_string = ""
 
-        self._http_url = 'http://{}:{}/jsonrpc'.format(host, port)
-        self._image_url = 'http://{}{}:{}/image'.format(
+        self._http_url = '{}:{}/jsonrpc'.format(host, port)
+        self._image_url = '{}{}:{}/image'.format(
```

Comment:
`self._image_url` is broken here. You'll end up with "user:passhttp://host". This is another place where keeping the host separate from the protocol is beneficial.